Rakefile

```diff
@@ -0,0 +1,60 @@
+# Rakefile
+# Copyright (C) 2018 Mike JS Choi
+# Copyright (C) 2018 VLC authors and VideoLAN
+# $Id$
+#
+# Authors: Mike JS. Choi <mkchoi212 # icloud.com>
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston MA 02110-1301, USA.
+#
+# ------------------------------------------------------------- Constants ------
+
+SDK_SIM = 'iphonesimulator11.3'
+SDK_SIM_DEST = "'platform=iOS Simulator,name=iPhone 7,OS=11.3'"
+
+SCHEME_IOS = 'MobileVLCKitTests'
+PROJECT_IOS = 'MobileVLCKit.xcodeproj'
+
+VLC_FLAGS_IOS = '-dva x86_64'
+
+# ----------------------------------------------------------------- Tasks ------
+
+desc 'Build VLCKit (iOS)'
+task 'build:vlckit:ios' do
+  puts 'Building VLCKit (iOS)'
+
+  plugin_file = 'Resources/MobileVLCKit/vlc-plugins-iPhone.h'
+  required_dirs = ['./libvlc/vlc/install-iPhoneSimulator', './libvlc/vlc/build-iPhoneSimulator']
+
+  if File.exist?(plugin_file) && dirs_exist?(required_dirs)
+    puts 'Found pre-existing build directory. Skipping build'
+  else
+    sh "./compileAndBuildVLCKit.sh #{VLC_FLAGS_IOS}"
+  end
+end
+
+desc 'Run MobileVLCKit tests'
+task 'test:ios' do
+  puts 'Running tests for MobileVLCKit'
+  sh "xcodebuild -project #{PROJECT_IOS} -scheme #{SCHEME_IOS} -sdk #{SDK_SIM} -destination #{SDK_SIM_DEST} test"
+end
+
+# ------------------------------------------------------------- Functions ------
+
+def dirs_exist?(directories)
+  directories.each do |dir|
+    return false unless Dir.exist?(dir)
```

Comment:
can't you just do this?
```
return Dir.exist?(dir)
```

Reply:
Are you talking about replacing the for-loop with `Dir.exist?(dir_1) && Dir.exist?(dir_2)`?

Reply:
no just replace the ```return false unless Dir.exist?(dir)``` with ```return Dir.exist?(dir)```

Reply:
If you do that, the for-loop would return after only evaluating the first element in the `directories` array. 

`return false unless Dir.exist?(dir)`, returns a `false` if and only if `Dir.exist?(dir)` evaluates to `false`.

If the loop doesn't return, the function implicitly returns a `true`

Reply:
I originally had something like 
```ruby
def dirs_exist?(directories)
  directories.each { |dir| 
    if Dir.exist?(x) == false { 
       return false
    }
  }
  return true 
```
but rubocop kept yelling at me until I got to what I have right now :D